modules/lib/lib-portal/src/main/resources/lib/xp/portal.ts

```diff
@@ -424,6 +424,25 @@ interface GetCurrentComponentHandler {
     execute(): object;
 }
 
+
+export type Component<
+  Config extends object = object,
+  Regions extends Record<string,Region> = Record<string,Region>
+> = {
+  config :Config
+  descriptor :string
+  path :string
+  type :'page'|'layout'|'part'
+  regions :Regions
```

Comment:
We don't use this kind of annotation in XP. It's always "`prop: propType`", not "`prop :propType`".

Reply:
I do this for multiple reasons. One is color in the editor.
![Screenshot 2022-09-07 at 11 52 33](https://user-images.githubusercontent.com/118847/188849085-c335945c-50c8-4cd7-9602-1dd1e93ef7b9.png)

Reply:
The top one just looks better... And is easier to read

Reply:
Also when searching I can differ between searching js objects, and ts types.

Reply:
Christian, we are not going to have two different types of annotation in one project. It's ok for Explorer since you used this as a standard there. Also you can surely setup highlighting differently in your IDE, so that's not an argument.

Reply:
I agree with @alansemenov . It's okay to have your preferences in one-man projects (that's what we all have). But XP is a large project and we have company code style here. As for the "just looks better" - it's a matter of taste and color scheme. In my color scheme I have a different colors for `:`, `?` and other symbols, because it's easier for ME to find those symbols, e.g.:
![image](https://user-images.githubusercontent.com/1847621/188855791-95b177bd-d486-47ca-8a41-5b2d05279478.png)

Reply:
Same goes for the spaces, semicolons. If you are using the Intellij IDEA, use this settings (though they might be kinds old)
https://github.com/enonic/dev-tools/blob/master/codestyles/idea/Enonic.xml

Reply:
Turns out switching syntax theme in Atom was easy (I probably already did before)
![Screenshot 2022-09-07 at 13 45 45](https://user-images.githubusercontent.com/118847/188870780-7f1b5a1b-e8b2-427d-8684-c9be5acc62a1.png)

Reply:
Great! Can we now change it in Explorer please? Just kidding 😀

Reply:
Fixed